### PR TITLE
Header 4.0.0, add contributor details in scaffold

### LINKED ISSUE #336: UI-Kit helper - Add questions for contributor name, email and website
Currently we have a function that creates a new module using the `helper.js` this finds and replaces items that have `[ something ]`.

We need to make sure it also does it for contributor name, email and website.

screen_shot_2018-02-12_at_3 10 24_pm

### LINKED ISSUE #362: Update the header component to be inline with the Australian Government branding guidelines
- [x] I’ve read and understood the [Contributing guidelines](./CONTRIBUTING.md) and have done my best effort to follow them.
- [x] I’ve read and agree to the [Code of Conduct](./CODE_OF_CONDUCT.md).
- [x] I’ve searched for any related issues and avoided creating a duplicate issue.

# What happened

The header component is currently missing details which makes it more useful. An update the header component is needed to bring it inline with the Australian Government branding guidelines.

We should include the generic Australian Government coat of arms so that there is a complete and functional header for the purposes of building a prototype. 

Eventually, we should allow agencies to supply their own coat of arms to replace the generic one, but this is dependent on #361

### PR DESCRIPTION
**Pull request by alex-page**
Closes #362 and closes #336

### CODE REVIEW

File: packages/header/CHANGELOG.md [hunk 1]
```diff
@@ -15,6 +15,7 @@
 
 ## Versions
 
+* [v4.0.0 - Added crest, and optional grid layout ](#v400)
```
Comment (pattyde) [inline packages/header/CHANGELOG.md]: I might be being a little picky, but could we rephrase this to say:

> Added placeholder for Australian Government logo and optional grid layout

File: packages/header/src/js/react.js [hunk 2]
```diff
@@ -18,47 +18,109 @@ import PropTypes from 'prop-types';
...
- * The header component
+ * The header brand component
  *
  * @param  {string}   title            - The title of the header
  * @param  {string}   level            - The tag level (<h1/> <h2/> etc), default: '1'
  * @param  {string}   subline          - An optional subline, optional
- * @param  {boolean}  hero             - The hero option, optional
- * @param  {boolean}  dark             - Add the dark variation class, optional
- * @param  {boolean}  alt              - Add the alt variation class, optional
+ * @param  {string}   crestImage       - An optional crest image, optional
```
Comment (pattyde) [inline packages/header/src/js/react.js]: ```* @param  {string}  logoImage       - Australian Government logo image, optional```

File: packages/header/src/js/react.js [hunk 3]
```diff
@@ -18,47 +18,109 @@ import PropTypes from 'prop-types';
...
+ * @param  {string}   crestAlt         - Crest alt tag
```
Comment (pattyde) [inline packages/header/src/js/react.js]: ```* @param  {string}   logoAlt         - Logo alt tag```

File: packages/header/src/js/react.js [hunk 4]
```diff
@@ -18,47 +18,109 @@ import PropTypes from 'prop-types';
...
  * @param  {node}     children         - The inside of this section
  * @param  {string}   className        - An additional class, optional
  * @param  {object}   attributeOptions - Any other attribute options
  */
-const AUheader = ({ title, level, subline, hero, dark, alt, children, className = '', ...attributeOptions }) => {
+export const AUheaderBrand = ({
+	title,
+	level,
+	subline,
+	crestImage,
```
Comment (pattyde) [inline packages/header/src/js/react.js]:
```logoImage,
logoAlt,```

File: packages/header/src/sass/_module.scss [hunk 5]
```diff
@@ -58,17 +50,121 @@
...
+	img {
+		display: block;
+		@include AU-space( max-width, $AU-logo-width );
+	}
+
+	img + .au-header__text {
+		@include AU-space( margin-top, 1unit );
+	}
+
+	@include AU-media( sm ){
```
Comment (pattyde) [inline packages/header/src/sass/_module.scss]: Missing space after `)`

```@include AU-media( sm ) {```

File: packages/header/src/sass/_globals.scss [hunk 6]
```diff
@@ -24,4 +24,5 @@ $dependencies: (
 // GLOBAL VARIABLES
 //--------------------------------------------------------------------------------------------------------------------------------------------------------------
 
-// Boy it sure is quiet here.
+$AU-logo-width: 12unit;
```
Comment (TrebBrennan) [inline packages/header/src/sass/_globals.scss]: Add `!default`

File: packages/header/src/sass/_module.scss [hunk 7]
```diff
@@ -22,7 +22,7 @@
  */
 .au-header {
 	@include AU-fontgrid( sm );
-	@include AU-space( padding, 1.5unit 0 2unit 0 ); // 0.5 for optical alignment
+	@include AU-space( padding, 1.5unit 0 ); // 0.5 for optical alignment
```
Comment (TrebBrennan) [inline packages/header/src/sass/_module.scss]: comment not needed

File: packages/header/src/sass/_print.scss [hunk 8]
```diff
@@ -12,4 +12,8 @@
 		color: #000 !important;
 		background: #fff !important;
 	}
+
+	.au-header-brand-logo {
```
Comment (TrebBrennan) [inline packages/header/src/sass/_print.scss]: Note this is not used in the screen CSS.
Should be `.au-header__brand img`

File: packages/header/tests/react/index.js [hunk 9]
```diff
@@ -1,100 +1,175 @@
...
-
-		<AUheader dark title="Page title" subline="Service sub-title" className="testing" />
-
-
-		<hr />
-		<h2>header with children</h2>
-
-		<AUheader title="Section title">
-			<button>Menu button?</button>
+		<AUheader className="au-grid" >
```
Comment (TrebBrennan) [inline packages/header/tests/react/index.js]: Space

File: packages/header/tests/site/index.html [hunk 10]
```diff
@@ -44,176 +44,361 @@
...
+				<h1 class="au-header__heading">Site title</h1>
+				<div class="au-header__subline">
+					Service description that could be a little longer Service description that could be a little longer Service description that could be a little longer
+				</div>
+			</div>
+		</a>
+	</header>
+
+
+	<h2>Hero header with service description with grid (E.g Home page)</h2>
```
Comment (pattyde) [inline packages/header/tests/site/index.html]: Should be "without grid"

File: packages/header/tests/site/index.html [hunk 11]
```diff
@@ -44,176 +44,361 @@
...
+	<header class="au-header au-header--alt" role="banner">
+		<a href="#" class="au-header__brand">
+			<h1 class="au-header__heading">Site title</h1>
+			<div class="au-header__subline">
+				Service description that could be a little longer
+			</div>
+		</a>
+	</header>
+
+	<h2>Header with service description with grid</h2>
```
Comment (pattyde) [inline packages/header/tests/site/index.html]: Should be "without grid"

File: packages/header/tests/site/index.html [hunk 12]
```diff
@@ -44,176 +44,361 @@
...
+
+	<h2>Logo Only (Eg. Agency Header)</h2>
+
+	<header class="au-header au-header--alt" role="banner">
+		<a href="#" class="au-header__brand">
+			<img class="au-header__brand-image" alt="Australian Government Crest" src="http://placehold.it/256x80">
+		</a>
+	</header>
+
+	<h2>Text only</h2>
```
Comment (pattyde) [inline packages/header/tests/site/index.html]: Worth mentioning that this is without `grid`?
